🐛 Bug Report: `yarn start --check` fails: "Expression expected"

### 📜 Description

The `--check` feature in Backstage CLI fails because ESLint is executed against an `index.html` (it tries to parse it as Javascript, but fails because it's HTML).

### 👍 Expected behavior

`app` should've been built successfully without errors

### 👎 Actual Behavior with Screenshots

```
$ yarn start --check
Loaded config from app-config.yaml
<i> [webpack-dev-server] Project is running at:
<i> [webpack-dev-server] Loopback: http://localhost:3000/, http://[::1]:3000/
<i> [webpack-dev-server] Content not from webpack is served from '/home/mitch/dev/backstage/packages/app/public' directory
<i> [webpack-dev-server] 404s will fallback to '/index.html'
<i> [webpack-dev-middleware] wait until bundle finished: /
yarn backstage-cli info
ERROR in   Error: Child compilation failed:
  [eslint] 
  /home/mitch/dev/backstage/packages/app/public/index.html
    1:0  error  Parsing error: Expression expected
  
  ✖ 1 problem (1 error, 0 warnings)
  
  
  - child-compiler.js:169 
    [backstage]/[html-webpack-plugin]/lib/child-compiler.js:169:18
  
  - Compiler.js:551 finalCallback
    [backstage]/[webpack]/lib/Compiler.js:551:5
  
  - Compiler.js:577 
    [backstage]/[webpack]/lib/Compiler.js:577:11
  
  - Compiler.js:1196 
    [backstage]/[webpack]/lib/Compiler.js:1196:17
  
  
  - task_queues:95 processTicksAndRejections
    node:internal/process/task_queues:95:5
  


1 ERROR in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)
webpack compiled with 2 errors
Type-checking in progress...

Error: ELOOP: too many symbolic links encountered, stat '/home/mitch/dev/backstage/packages/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app/node_modules/techdocs-cli-embedded-app'

```

### 👟 Reproduction steps

1. `git checkout c010a30fe97147b925cc1ef4276291eb17c95169` (current `master`)
2. `yarn`
3. `cd packages/app`
4. `yarn start --check`

### 📃 Provide the context for the Bug.

I set up a custom build that calls `serveBundle()` (from `@backstage/cli` directly) and was confused why having `checksEnabled: true` broke things for me. 

I wanted to compare against a "successful" `checksEnabled: true` configuration, which was why I tried it in `packages/app` - and am reporting this issue because I'm guessing it should work for that subproject.

I can work around this by leaving `checksEnabled: false` (which is the default).

### 🖥️ Your Environment

```
$ yarn backstage-cli info
OS:   Linux 6.1.9-200.fc37.x86_64 - linux/x64
node: v18.12.1
yarn: 3.2.3
cli:  0.22.2-next.1 (local)
backstage:  N/A

Dependencies:
  @backstage/app-defaults                                        0.0.0-use.local
  @backstage/backend-app-api                                     0.0.0-use.local
  @backstage/backend-common                                      0.0.0-use.local
  @backstage/backend-defaults                                    0.0.0-use.local
  @backstage/backend-dev-utils                                   0.0.0-use.local
  @backstage/backend-plugin-api                                  0.0.0-use.local
  @backstage/backend-tasks                                       0.0.0-use.local
  @backstage/backend-test-utils                                  0.0.0-use.local
  @backstage/catalog-client                                      1.3.0, 0.0.0-use.local
  @backstage/catalog-model                                       1.1.5, 0.0.0-use.local
  @backstage/cli-common                                          0.0.0-use.local
  @backstage/cli                                                 0.0.0-use.local
  @backstage/codemods                                            0.0.0-use.local
  @backstage/config-loader                                       0.0.0-use.local
  @backstage/config                                              0.0.0-use.local
  @backstage/core-app-api                                        0.0.0-use.local
  @backstage/core-components                                     0.12.3, 0.0.0-use.local
  @backstage/core-plugin-api                                     0.0.0-use.local
  @backstage/create-app                                          0.0.0-use.local
  @backstage/dev-utils                                           0.0.0-use.local
  @backstage/errors                                              0.0.0-use.local
  @backstage/eslint-plugin                                       0.0.0-use.local
  @backstage/integration-aws-node                                0.0.0-use.local
  @backstage/integration-react                                   1.1.9, 0.0.0-use.local
  @backstage/integration                                         0.0.0-use.local
  @backstage/plugin-adr-backend                                  0.0.0-use.local
  @backstage/plugin-adr-common                                   0.0.0-use.local
  @backstage/plugin-adr                                          0.0.0-use.local
  @backstage/plugin-airbrake-backend                             0.0.0-use.local
  @backstage/plugin-airbrake                                     0.0.0-use.local
  @backstage/plugin-allure                                       0.0.0-use.local
  @backstage/plugin-analytics-module-ga                          0.0.0-use.local
  @backstage/plugin-apache-airflow                               0.0.0-use.local
  @backstage/plugin-api-docs-module-protoc-gen-doc               0.0.0-use.local
  @backstage/plugin-api-docs                                     0.0.0-use.local
  @backstage/plugin-apollo-explorer                              0.0.0-use.local
  @backstage/plugin-app-backend                                  0.0.0-use.local
  @backstage/plugin-auth-backend                                 0.0.0-use.local
  @backstage/plugin-auth-node                                    0.0.0-use.local
  @backstage/plugin-azure-devops-backend                         0.0.0-use.local
  @backstage/plugin-azure-devops-common                          0.0.0-use.local
  @backstage/plugin-azure-devops                                 0.0.0-use.local
  @backstage/plugin-azure-sites-backend                          0.0.0-use.local
  @backstage/plugin-azure-sites-common                           0.0.0-use.local
  @backstage/plugin-azure-sites                                  0.0.0-use.local
  @backstage/plugin-badges-backend                               0.0.0-use.local
  @backstage/plugin-badges                                       0.0.0-use.local
  @backstage/plugin-bazaar-backend                               0.0.0-use.local
  @backstage/plugin-bazaar                                       0.0.0-use.local
  @backstage/plugin-bitbucket-cloud-common                       0.0.0-use.local
  @backstage/plugin-bitrise                                      0.0.0-use.local
  @backstage/plugin-catalog-backend-module-aws                   0.0.0-use.local
  @backstage/plugin-catalog-backend-module-azure                 0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-cloud       0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket-server      0.0.0-use.local
  @backstage/plugin-catalog-backend-module-bitbucket             0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gerrit                0.0.0-use.local
  @backstage/plugin-catalog-backend-module-github                0.0.0-use.local
  @backstage/plugin-catalog-backend-module-gitlab                0.0.0-use.local
  @backstage/plugin-catalog-backend-module-incremental-ingestion 0.0.0-use.local
  @backstage/plugin-catalog-backend-module-ldap                  0.0.0-use.local
  @backstage/plugin-catalog-backend-module-msgraph               0.0.0-use.local
  @backstage/plugin-catalog-backend-module-openapi               0.0.0-use.local
  @backstage/plugin-catalog-backend                              0.0.0-use.local
  @backstage/plugin-catalog-common                               1.0.10, 0.0.0-use.local
  @backstage/plugin-catalog-graph                                0.0.0-use.local
  @backstage/plugin-catalog-graphql                              0.0.0-use.local
  @backstage/plugin-catalog-import                               0.0.0-use.local
  @backstage/plugin-catalog-node                                 0.0.0-use.local
  @backstage/plugin-catalog-react                                1.2.4, 0.0.0-use.local
  @backstage/plugin-catalog                                      0.0.0-use.local
  @backstage/plugin-cicd-statistics-module-gitlab                0.0.0-use.local
  @backstage/plugin-cicd-statistics                              0.0.0-use.local
  @backstage/plugin-circleci                                     0.0.0-use.local
  @backstage/plugin-cloudbuild                                   0.0.0-use.local
  @backstage/plugin-code-climate                                 0.0.0-use.local
  @backstage/plugin-code-coverage-backend                        0.0.0-use.local
  @backstage/plugin-code-coverage                                0.0.0-use.local
  @backstage/plugin-codescene                                    0.0.0-use.local
  @backstage/plugin-config-schema                                0.0.0-use.local
  @backstage/plugin-cost-insights-common                         0.0.0-use.local
  @backstage/plugin-cost-insights                                0.0.0-use.local
  @backstage/plugin-dynatrace                                    0.0.0-use.local
  @backstage/plugin-entity-validation                            0.0.0-use.local
  @backstage/plugin-events-backend-module-aws-sqs                0.0.0-use.local
  @backstage/plugin-events-backend-module-azure                  0.0.0-use.local
  @backstage/plugin-events-backend-module-bitbucket-cloud        0.0.0-use.local
  @backstage/plugin-events-backend-module-gerrit                 0.0.0-use.local
  @backstage/plugin-events-backend-module-github                 0.0.0-use.local
  @backstage/plugin-events-backend-module-gitlab                 0.0.0-use.local
  @backstage/plugin-events-backend-test-utils                    0.0.0-use.local
  @backstage/plugin-events-backend                               0.0.0-use.local
  @backstage/plugin-events-node                                  0.0.0-use.local
  @backstage/plugin-explore-backend                              0.0.0-use.local
  @backstage/plugin-explore-common                               0.0.0-use.local
  @backstage/plugin-explore-react                                0.0.0-use.local
  @backstage/plugin-explore                                      0.0.0-use.local
  @backstage/plugin-firehydrant                                  0.0.0-use.local
  @backstage/plugin-fossa                                        0.0.0-use.local
  @backstage/plugin-gcalendar                                    0.0.0-use.local
  @backstage/plugin-gcp-projects                                 0.0.0-use.local
  @backstage/plugin-git-release-manager                          0.0.0-use.local
  @backstage/plugin-github-actions                               0.0.0-use.local
  @backstage/plugin-github-deployments                           0.0.0-use.local
  @backstage/plugin-github-issues                                0.0.0-use.local
  @backstage/plugin-github-pull-requests-board                   0.0.0-use.local
  @backstage/plugin-gitops-profiles                              0.0.0-use.local
  @backstage/plugin-gocd                                         0.0.0-use.local
  @backstage/plugin-graphiql                                     0.0.0-use.local
  @backstage/plugin-graphql-backend                              0.0.0-use.local
  @backstage/plugin-graphql-voyager                              0.0.0-use.local
  @backstage/plugin-home                                         0.4.30, 0.0.0-use.local
  @backstage/plugin-ilert                                        0.0.0-use.local
  @backstage/plugin-jenkins-backend                              0.0.0-use.local
  @backstage/plugin-jenkins-common                               0.0.0-use.local
  @backstage/plugin-jenkins                                      0.0.0-use.local
  @backstage/plugin-kafka-backend                                0.0.0-use.local
  @backstage/plugin-kafka                                        0.0.0-use.local
  @backstage/plugin-kubernetes-backend                           0.0.0-use.local
  @backstage/plugin-kubernetes-common                            0.0.0-use.local
  @backstage/plugin-kubernetes                                   0.0.0-use.local
  @backstage/plugin-lighthouse-backend                           0.0.0-use.local
  @backstage/plugin-lighthouse-common                            0.0.0-use.local
  @backstage/plugin-lighthouse                                   0.0.0-use.local
  @backstage/plugin-microsoft-calendar                           0.0.0-use.local
  @backstage/plugin-newrelic-dashboard                           0.0.0-use.local
  @backstage/plugin-newrelic                                     0.0.0-use.local
  @backstage/plugin-org-react                                    0.0.0-use.local
  @backstage/plugin-org                                          0.0.0-use.local
  @backstage/plugin-pagerduty                                    0.0.0-use.local
  @backstage/plugin-periskop-backend                             0.0.0-use.local
  @backstage/plugin-periskop                                     0.0.0-use.local
  @backstage/plugin-permission-backend                           0.0.0-use.local
  @backstage/plugin-permission-common                            0.0.0-use.local
  @backstage/plugin-permission-node                              0.0.0-use.local
  @backstage/plugin-permission-react                             0.0.0-use.local
  @backstage/plugin-playlist-backend                             0.0.0-use.local
  @backstage/plugin-playlist-common                              0.0.0-use.local
  @backstage/plugin-playlist                                     0.0.0-use.local
  @backstage/plugin-proxy-backend                                0.0.0-use.local
  @backstage/plugin-rollbar-backend                              0.0.0-use.local
  @backstage/plugin-rollbar                                      0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-cookiecutter       0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-rails              0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-sentry             0.0.0-use.local
  @backstage/plugin-scaffolder-backend-module-yeoman             0.0.0-use.local
  @backstage/plugin-scaffolder-backend                           0.0.0-use.local
  @backstage/plugin-scaffolder-common                            0.0.0-use.local
  @backstage/plugin-scaffolder-node                              0.0.0-use.local
  @backstage/plugin-scaffolder-react                             0.0.0-use.local
  @backstage/plugin-scaffolder                                   0.0.0-use.local
  @backstage/plugin-search-backend-module-elasticsearch          0.0.0-use.local
  @backstage/plugin-search-backend-module-pg                     0.0.0-use.local
  @backstage/plugin-search-backend-node                          0.0.0-use.local
  @backstage/plugin-search-backend                               0.0.0-use.local
  @backstage/plugin-search-common                                0.0.0-use.local
  @backstage/plugin-search-react                                 1.4.0, 0.0.0-use.local
  @backstage/plugin-search                                       0.0.0-use.local
  @backstage/plugin-sentry                                       0.0.0-use.local
  @backstage/plugin-shortcuts                                    0.0.0-use.local
  @backstage/plugin-sonarqube-backend                            0.0.0-use.local
  @backstage/plugin-sonarqube-react                              0.0.0-use.local
  @backstage/plugin-sonarqube                                    0.0.0-use.local
  @backstage/plugin-splunk-on-call                               0.0.0-use.local
  @backstage/plugin-stack-overflow-backend                       0.0.0-use.local
  @backstage/plugin-stack-overflow                               0.1.10, 0.0.0-use.local
  @backstage/plugin-tech-insights-backend-module-jsonfc          0.0.0-use.local
  @backstage/plugin-tech-insights-backend                        0.0.0-use.local
  @backstage/plugin-tech-insights-common                         0.0.0-use.local
  @backstage/plugin-tech-insights-node                           0.0.0-use.local
  @backstage/plugin-tech-insights                                0.0.0-use.local
  @backstage/plugin-tech-radar                                   0.0.0-use.local
  @backstage/plugin-techdocs-addons-test-utils                   0.0.0-use.local
  @backstage/plugin-techdocs-backend                             0.0.0-use.local
  @backstage/plugin-techdocs-module-addons-contrib               0.0.0-use.local
  @backstage/plugin-techdocs-node                                0.0.0-use.local
  @backstage/plugin-techdocs-react                               0.0.0-use.local
  @backstage/plugin-techdocs                                     0.0.0-use.local
  @backstage/plugin-todo-backend                                 0.0.0-use.local
  @backstage/plugin-todo                                         0.0.0-use.local
  @backstage/plugin-user-settings-backend                        0.0.0-use.local
  @backstage/plugin-user-settings                                0.0.0-use.local
  @backstage/plugin-vault-backend                                0.0.0-use.local
  @backstage/plugin-vault                                        0.0.0-use.local
  @backstage/plugin-xcmetrics                                    0.0.0-use.local
  @backstage/release-manifests                                   0.0.0-use.local
  @backstage/repo-tools                                          0.0.0-use.local
  @backstage/test-utils                                          0.0.0-use.local
  @backstage/theme                                               0.0.0-use.local
  @backstage/types                                               0.0.0-use.local
  @backstage/version-bridge                                      0.0.0-use.local
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

--- a/.changeset/light-sheep-trade.md
+++ b/.changeset/light-sheep-trade.md
@@ -1,0 +1,5 @@
+---
+'@backstage/cli': patch
+---
+
+Updated the ESLint plugin configuration that is enabled through `yarn start --check` to only pick up valid source files.

--- a/packages/cli/src/lib/bundler/config.ts
+++ b/packages/cli/src/lib/bundler/config.ts
@@ -100,7 +100,7 @@ export async function createConfig(
       }),
       new ESLintPlugin({
         context: paths.targetPath,
-        files: ['**', '!**/__tests__/**', '!**/?(*.)(spec|test).*'],
+        files: ['**/*.(ts|tsx|mts|cts|js|jsx|mjs|cjs)'],
       }),
     );
   }
@@ -325,7 +325,7 @@ export async function createBackendConfig(
               typescript: { configFile: paths.targetTsConfig },
             }),
             new ESLintPlugin({
-              files: ['**', '!**/__tests__/**', '!**/?(*.)(spec|test).*'],
+              files: ['**/*.(ts|tsx|mts|cts|js|jsx|mjs|cjs)'],
             }),
           ]
         : []),

